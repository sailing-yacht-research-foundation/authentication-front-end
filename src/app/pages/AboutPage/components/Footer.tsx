import React from 'react';
import styled from 'styled-components';
import { Typography, Button, Row, Col, Divider } from 'antd';
import { Link } from 'react-router-dom';
import { media } from 'styles/media';
import { useTranslation } from 'react-i18next';
import { translations } from 'locales/translations';
import { ReactComponent as Logo } from './assets/logo-light.svg';

export const Footer = (props) => {

    const { t } = useTranslation();

    return (
        <Wrapper>
            <DeveloperSectionContainer>
                <Typography.Title className="section-header-text">
                    {t(translations.home_page.footer.title)}
                </Typography.Title>

                <DeveloperDescription>
                    {t(translations.home_page.footer.description)}
                </DeveloperDescription>

                <Button size={'large'} className="syrf-button" style={{ marginTop: '20px', marginBottom: '20px' }}>{t(translations.home_page.footer.developer_button)}</Button>
            </DeveloperSectionContainer>

            <SYRFFooterContainer>
                <Row justify="space-between">
                    <Col lg={{
                        offset: 3,
                        span: 4
                    }}
                        xs={{
                            span: 24
                        }}
                    >
                        <Logo style={{ marginBottom: '15px' }}/>
                        <SYRFDescription>{t(translations.home_page.footer.syrf_description)}</SYRFDescription>
                    </Col>

                    <Col
                        lg={{
                            span: 8,
                            offset: 8
                        }}
                        xs={{
                            span: 24
                        }}
                    >
                        <StyledRow justify="end" align="middle" style={{ marginTop: '50px' }}>
                            <StyledLink to="/">SYRF</StyledLink>
                        </StyledRow>

                        <StyledRow justify="end" align="middle" style={{ marginTop: '10px' }}>
                            <StyledLink to="privacy-policy">{t(translations.home_page.footer.policy_privacy_link)}</StyledLink>
                        </StyledRow>

                        <StyledRow justify="end" align="middle" style={{ marginTop: '10px' }}>
                            <StyledLink to="eula">{t(translations.home_page.footer.eula_link)}</StyledLink>
                        </StyledRow>
                    </Col>
                </Row>

                <Divider />

                <FooterTextCopyRight>{t(translations.home_page.footer.copyright)}</FooterTextCopyRight>
            </SYRFFooterContainer>
        </Wrapper>
    )
}

const Wrapper = styled.div`
    margin-top: 50px;
    display: flex;
    justify-content:center;
    align-items: center;
    flex-direction: column;
    // background-color: #001529;
    text-align: center;
`;

const FooterTextCopyRight = styled.div`
   margin-top: 22px;
   margin-bottom: 22px;
   color: #fff;
   font-size: 0.8rem;
   text-align: center;
`;

const DeveloperSectionContainer = styled.div`
    background: #EBEFF7;
    width: 100%;
    text-align:center;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 86px 15px;
`;

const DeveloperDescription = styled.div`
    font-size: 14px;
    font-weight: 400;
    max-width: 805px;
    margin: 15px 0;
`;

const SYRFFooterContainer = styled.div`
    background: #0C4983;
    width: 100%;
    padding: 0 15px;
    padding-top: 65px;
`;

const SYRFDescription = styled.p`
    font-weight: 400;
    color: #fff;

    ${media.large`
        text-align: left !important;
    `};

    ${media.medium`
        text-align: left !important;
    `};

    ${media.small`
        text-align:center;
    `};
`;

const StyledLink = styled(Link)`
    font-size: 24px;
    font-weight: bold;
    color: #fff;

    ${media.large`
        margin-right: 170px;
        text-align: right;
    `};

    ${media.medium`
        margin-right: 120px;
        text-align: right;
    `};
`;

const StyledRow = styled(Row)`
        justify-content: center;

    ${media.large`
        justify-content: flex-end;
    `};

    ${media.medium`
        justify-content: flex-end;
    `};

    ${media.small`
        justify-contents: center;
    `};
`