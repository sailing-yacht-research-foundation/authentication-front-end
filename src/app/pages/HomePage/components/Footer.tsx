import React from 'react';
import styled from 'styled-components';
import { Typography, Button, Row, Col, Divider } from 'antd';
import { Logo } from 'app/components/NavBar/Logo';
import { Link } from 'react-router-dom';

export const Footer = (props) => {
    return (
        <Wrapper>
            <Typography.Title className="text-white" style={{ marginTop: '25px' }} level={3}>
                Are you a developer who wants to use My Sailing Profile?
            </Typography.Title>
            <Button size={'large'} className="syrf-button" style={{ marginTop: '20px', marginBottom: '20px' }}>Developers</Button>
            <Row justify="center" align="middle">
                <Col lg={16}>
                    <Row>
                        <Col lg={5}>
                            <Logo type="light" style={{ marginBottom: '30px' }} />
                            <span className="text-white">Lorem ipsum dolor sit amet consectetur adipisicing elit. Odit veritatis, in consequatur voluptatum velit ea aperiam dolore magni voluptates alias id delectus suscipit quae nulla! Delectus provident fuga doloribus vel.</span>
                        </Col>
                        <Col lg={14} xs={24} style={{ marginTop: '50px' }}>
                            <Row justify="center" align="middle" style={{ marginTop: '10px' }}>
                                <a className="text-white" href="https://syrf.io">SYRF</a>
                            </Row>
                            <Row justify="center" align="middle" style={{ marginTop: '10px' }}>
                                <Link className="text-white" to="https://syrf.io">Privacy Policy</Link>
                            </Row>
                            <Row justify="center" align="middle" style={{ marginTop: '10px' }}>
                                <Link className="text-white" to="https://syrf.io">End user license agreement</Link>
                            </Row>
                        </Col>
                    </Row>
                </Col>
            </Row>
            <Divider/>
            <Row>
                <FooterTextCopyRight>Copyright (c) 2021 Sailing Yacht Research Foundation</FooterTextCopyRight>
            </Row>
        </Wrapper>
    )
}

const Wrapper = styled.div`
    margin-top: 50px;
    display: flex;
    justify-content:center;
    align-items: center;
    flex-direction: column;
    background-color: #001529;
    padding: 30px 15px;
    text-align: center;
`

const FooterTextCopyRight = styled.div`
   margin-top: 30px;
   color: #fff;
   font-size: 0.8rem;
`