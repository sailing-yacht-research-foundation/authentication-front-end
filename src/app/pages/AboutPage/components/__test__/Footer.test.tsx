import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Footer } from '../Footer';

const shallowRenderer = createRenderer();

describe('<Footer />', () => {
  it('should render and matchs the snapshot', () => {
    shallowRenderer.render(<Footer />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});