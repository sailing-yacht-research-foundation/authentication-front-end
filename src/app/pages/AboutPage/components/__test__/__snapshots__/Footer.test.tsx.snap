// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render and matchs the snapshot 1`] = `
<styled.div>
  <styled.div>
    <Title
      className="section-header-text"
    >
      Are you a developer who wants to use My Sailing Profile?
    </Title>
    <styled.div>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting.
    </styled.div>
    <Button
      className="syrf-button"
      size="large"
      style={
        Object {
          "marginBottom": "20px",
          "marginTop": "20px",
        }
      }
    >
      Developers
    </Button>
  </styled.div>
  <styled.div>
    <Row
      justify="space-between"
    >
      <Col
        lg={
          Object {
            "offset": 3,
            "span": 4,
          }
        }
        xs={
          Object {
            "span": 24,
          }
        }
      >
        <ForwardRef(SvgLogoLight)
          style={
            Object {
              "marginBottom": "15px",
            }
          }
        />
        <styled.p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Odit veritatis, in consequatur voluptatum velit ea aperiam dolore magni voluptates alias id delectus suscipit quae nulla! Delectus provident fuga doloribus vel.
        </styled.p>
      </Col>
      <Col
        lg={
          Object {
            "offset": 8,
            "span": 8,
          }
        }
        xs={
          Object {
            "span": 24,
          }
        }
      >
        <Styled(Row)
          align="middle"
          justify="end"
          style={
            Object {
              "marginTop": "50px",
            }
          }
        >
          <Styled(Link)
            to="/"
          >
            SYRF
          </Styled(Link)>
        </Styled(Row)>
        <Styled(Row)
          align="middle"
          justify="end"
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <Styled(Link)
            to="privacy-policy"
          >
            Privacy Policy
          </Styled(Link)>
        </Styled(Row)>
        <Styled(Row)
          align="middle"
          justify="end"
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <Styled(Link)
            to="eula"
          >
            End User License Agreement
          </Styled(Link)>
        </Styled(Row)>
      </Col>
    </Row>
    <Divider />
    <styled.div>
      Copyright (c) 2021 Sailing Yacht Research Foundation
    </styled.div>
  </styled.div>
</styled.div>
`;
